src/validate.ts

```diff
@@ -0,0 +1,129 @@
+'use strict';
+
+const Ajv = require('ajv'); 
+const ajv = new Ajv();
+const chai = require('chai');
+chai.use(require('chai-json-schema-ajv'));
+
+//calling the JSON schemas to throw assertion errors
+const sync_response_schema = require('../intents/sync.response.schema.json');
+const apiresponse = require('./apiresponse.json');
+
+
+var testsPassed = 0
+var testsFailed = 0
+var totalTests = 0
+
+/**
+ * Generate requests to validate the responses later on  
+*/
+
+function generateRandom(min, max){
+    //helper function for obtaining a random requestId
+    return Math.floor(Math.random() * (max-min) + min).toString();
+}
+
+
+function generateSyncReq(){
+    const requestRandId = generateRandom(100,999);
+    return {
+    "requestId": requestRandId,
+    "inputs": [{
+      "intent": "action.devices.SYNC"
+        }]
+    }
+}
+
+const sync_request = generateSyncReq();
+const requestRandId = sync_request["requestId"];
+
+function generateQueryReq(deviceID){
+    
+    const intent = "action.devices.QUERY";
+    var queryReq = {};
+    queryReq.requestId = requestRandId;
+    queryReq.inputs = [];
+    queryReq.inputs.unshift({});
+    queryReq.inputs[0]["intent"] = intent;
+    queryReq.payload = {};
+    queryReq.payload.devices = [];
+    queryReq.payload.devices.unshift({})
+    queryReq.payload.devices[0]["id"] = deviceID;
+
+    
+    
+    //  for (var i = 0; i <deviceID.length; i++){
+    //         queryReq.payload.devices[i]["id"] = deviceID[i];
+    //     }
+    //add custom data
+    
+    return queryReq;
+    
+    
+}
+
+function add(num){
+    var json = [{"cool": num}];
+    json.coolness = 34.33;
+    return json
+}
+
+console.log(add(4));
+console.log(add("5"));
+
+
+const query_request = generateQueryReq("123")["payload"]["devices"];
+
+console.log(query_request)
+// const deviceInfo = (query_request["inputs"][0]["payload"]);
+
+function generateExecuteReq(traits, attributes){
+    return {
+        "requestId": requestId,
+        "inputs": [{
+          "intent": "action.devices.EXECUTE",
+          "payload": {
+            "commands": deviceInfo,
+              "execution": [{
+                "command": traits,
+                "params": {
+                  "on": true
+                }
+              }]
+          }
+        }]
+      }
+    }
+
+
+function generateDisconnectReq(){
+    return {
+        "requestId": sync_request["requestId"],
+        "inputs": [{
+          "intent": "action.devices.DISCONNECT",
+        }]
+    }
+}
+
+
+function validate(schema, apiresponse){
```

Comment:
Can we omit this function in this pull request and do it in a follow-up?

Reply:
Yes, removed for now